main.tf

```diff
@@ -146,7 +146,7 @@ resource "aws_ecs_service" "service" {
   }
 
   load_balancer {
-    container_name   = "${var.name_prefix}"
+    container_name   = "${var.container_name == "" ? var.name_prefix : var.container_name}"
```

Comment:
same here if var.container_name is itself interpolated before it gets here this line will fail.